src/periphery/MarketLens.sol

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.9;
+
+// Uncomment this line to use console.log
+
+import "interfaces/IBentoBoxV1.sol";
+import "interfaces/ICauldronV2.sol";
+import {IERC20, IStrictERC20} from "BoringSolidity/interfaces/IERC20.sol";
+import "utils/CauldronLib.sol";
+
+contract MarketLens {
+    address payable public owner;
+    uint8 constant DECIMALS = 18;
+
+    constructor() {
+        owner = payable(msg.sender);
+    }
+
+    function getInterestPerYear(ICauldronV2 cauldron) external view returns (uint64 interestPerYear) {
+        (, , uint64 interestPerSecond) = cauldron.accrueInfo();
+        interestPerYear = CauldronLib.getInterestPerYearFromInterestPerSecond(interestPerSecond);
+    }
+
+    function getMaxBorrowForCauldronV2(ICauldronV2 cauldron) external view returns (uint256 maxBorrow) {
+        IBentoBoxV1 bentoBox = IBentoBoxV1(cauldron.bentoBox());
+        IERC20 mim = IERC20(cauldron.magicInternetMoney());
+        uint256 poolBalance = bentoBox.balanceOf(mim, address(cauldron));
+        maxBorrow = bentoBox.toAmount(mim, poolBalance, false);
+    }
+
+    function getMaxBorrowForCauldronV3(ICauldronV3 cauldron) external returns (uint256 maxBorrow) {
+        IBentoBoxV1 bentoBox = IBentoBoxV1(cauldron.bentoBox());
+        IERC20 mim = IERC20(cauldron.magicInternetMoney());
+        uint256 poolBalance = bentoBox.balanceOf(mim, address(cauldron));
+        uint256 mimInBentoBox = bentoBox.toAmount(mim, poolBalance, false);
+        uint256 userBorrowLimit = getUserBorrowLimit(cauldron);
+        maxBorrow = min(userBorrowLimit, mimInBentoBox);
+    }
+
+    // Why does this modify state? Want to be able to use view but cannot.
+    function getUserBorrowLimit(ICauldronV3 cauldron) public returns (uint256 userBorrowLimit) {
+        (uint256 totalLimit, uint256 borrowPartPerAddress) = cauldron.borrowLimit();
+        userBorrowLimit = min(totalLimit, borrowPartPerAddress);
+    }
+
+    function getTotalMimBorrowed(ICauldronV2 cauldron) external view returns (uint256 totalMimBorrowed) {
+        Rebase memory totalBorrow = cauldron.totalBorrow();
+        totalMimBorrowed = totalBorrow.elastic;
+    }
+
+    function getTvl(ICauldronV2 cauldron) external view returns (uint256 tvl) {
+        IBentoBoxV1 bentoBox = IBentoBoxV1(cauldron.bentoBox());
+        uint256 totalTokensDeposited = bentoBox.toAmount(cauldron.collateral(), cauldron.totalCollateralShare(), false);
+        tvl = (totalTokensDeposited * 10 ** DECIMALS) / getOracleExchangeRate(cauldron);
+    }
+
+    function getOracleExchangeRate(ICauldronV2 cauldron) public view returns (uint256 exchangeRate) {
+        IOracle oracle = IOracle(cauldron.oracle());
+        bytes memory oracleData = cauldron.oracleData();
+        exchangeRate = oracle.peekSpot(oracleData);
+    }
+
+    // How can I get the LIQUIDATION_MULTIPLIER constant?
```

Comment:
I noticed the LIQUIDATION_MULTIPLIER is not in the interface... Why not?

Reply:
it is on ICauldronV2.sol

Reply:
Oh it's because you need to access it like LIQUIDATION_MULTIPLIER(), like a function call

Reply:
This is what I get...

<img width="734" alt="Screenshot 2023-02-03 at 7 51 43 AM" src="https://user-images.githubusercontent.com/118562806/216647285-409d9234-5dad-419e-a57e-95248a99352b.png">

I also still don't see that function inside the ICauldronV2.sol... I see it listed as a public constant on the Cauldron contract, but not on the interface. Is that a problem?

Reply:
ah, i modified the interface for the magicGLP PR so you can copy paste this in your branch and that shouldn't create conflicts

https://github.com/Abracadabra-money/abracadabra-money-foundry/blob/glp-auto-compounding-cauldron/src/interfaces/ICauldronV2.sol

Reply:
Awesome. Thanks!